docs/types.rst

```diff
@@ -488,6 +507,11 @@ omitted. Note that this only applies to function types. Visibility has
 to be specified explicitly for functions defined in contracts, they
 do not have a default.
 
+Conversions:
+
+A value of external function type can be explicitly converted to ``address``
+resulting in the address of the contract of the function.
```

Comment:
Hm, can function types be payable and if so do they cast to `address payable`?

Reply:
They should always cast to `address`. The fact that one function in the contract is payable does not mean that the contract will have a payable fallback function.

Reply:
I'll check if we have a test.

Reply:
It seems this was a bug. Created PR: https://github.com/ethereum/solidity/pull/5030